proposals/Scoped-Custom-Element-Registries.md

```diff
@@ -119,6 +121,23 @@ As a result, it must limit constructors by default to only looking up registrati
 
 This poses a limitation for authors trying to use the constructor to create new elements associated to scoped registries but not registered as global. More flexibility can be analyzed post MVP, for now, a user-land abstraction can help by keeping track of the constructor and its respective registry.
 
+### Note on light DOM custom elements
+
+Custom elements that use light DOM (i.e. that don't call `this.attachShadow()`) may be scoped, but they must be scoped within a shadow root. For example:
+
+```html
+<body>
+  <shadow-element>
+    #shadow-root (registry=myCustomRegistry)
+      <light-element>
+        <div>Light DOM</div>
+      </light-element>
+  </shadow-element>
+</body>
+```
+
+In the above example, `<light-element>` is scoped within the shadow root of its containing `<shadow-element>`, whereas `<shadow-element>` is defined at the global document level.
```

Comment:
Should we reflect that scope could be more flexible and tied not just to shadowRoot?

There is a scenario when `light-element` would have declarative custom element definition inside which would be logical to scope to the parent (or somehow named scope) instead whole page. 


```html
<body>
  <light-element>
    #light-root (registry=myCustomRegistry)
      <definition tag="custom-element>...</definition><!-- syntax is not clear as of now -->
      <custom-element>
        <div>Light DOM</div>
      </custom-element>
  </light-element>
</body>
```

Reply:
I'm not sure I understand this. Declarative custom element definitions do not exist yet, and there's no such thing as a light-root.

If `<light-element>` wants its own scope it'll have to programmatically create one and ensure that an elements it create use scoped instantiation APIs, which right now are only available on ShadowRoot, though we could conceivably also add them to CustomElementRegistry.

Reply:
I was under the impression that Declarative Shadow DOM was out-of-scope for this proposal... Although there is some reference to it [here](https://github.com/WICG/webcomponents/blob/gh-pages/proposals/Scoped-Custom-Element-Registries.md#intersection-with-declarative-shadow-root).

Reply:
Declarative Shadow DOM has its sibling and evolution as [Declarative Custom Element](https://github.com/w3c/webcomponents-cg/issues/32)  which could use shadow but also could live without, both in a baking now, not yet a proposal but already have POCs.

Reply:
Declarative shadow DOM has nothing to do with declarative custom elements. They are unrelated conceptually, as declarative custom elements would _not_ use declarative shadow DOM, and are at very, very different stages of proposal and implementation.